sync/base.py

```diff
@@ -1,15 +1,17 @@
+from __future__ import absolute_import
 import json
 import weakref
 from collections import defaultdict, Mapping
 
 import git
 import pygit2
 
-import log
-import commit as sync_commit
-from env import Environment
-from lock import MutGuard, RepoLock, mut, constructor
-from repos import pygit2_get
+from . import log
+from . import commit as sync_commit
+from .env import Environment
+from .lock import MutGuard, RepoLock, mut, constructor
+from .repos import pygit2_get
```

Comment:
The `from __future__ import absolute_imports` forces you into the Python 3 import syntax, even though as you say the cleaned up imports are relative. I think this could be something to clean up in the future rather than in in this PR.